Angular/Angular guidelines and best practices/Testing.md

```diff
@@ -531,6 +531,96 @@ beforeEach(async(() => {
+
+The downside of using a host component is that it is a bit boilerplate-y - you will have to bind potentially many inputs and outputs between the host and child components.
+
+We recommend creating a host component, especially for cases where you need to test input/output interaction.
+
+```typescript
+@Component({
+  selector: 'counter'
+  template: `
+  {{ value }}
+  <button class="counter-button" (click)="onCounterButtonClick()">Increase</button>
+  `
+})
+export class CounterComponent {
+  @Input() public value: number = 0;
+  @Output() public valueChange = new EventEmitter<number>();
+
+  public onCounterButtonClick(): void {
+    this.value++;
+    this.valueChange.emit(this.value);
+  }
+}
+```
+
+```typescript
+@Component({
+  selector: 'counter-host'
+  template: `
+  <counter
+    [value]="value"
+    (valueChange)="onValueChange()
+  >
+  </counter>
+  `
+})
+class CounterHostComponent {
+  public value: number = 0; // Notice that this doesn't have to be an @Input
+
+  public onValueChange(newValue: number) { }
+}
+
+describe('CounterComponent', () => {
+  let fixture: ComponentFixture<CounterHostComponent>;
+  let hostComponent: CounterHostComponent;
+  let component: CounterComponent;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [CounterComponent, CounterHostComponent],
+      imports: [...],
+      providers: [...],
+    }).compileComponents();
+  })
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(CounterHostComponent);
+    fixture.detectChanges();
+
+    hostComponent = fixture.componentInstance;
+    component = fixture.debugElement.query(By.css('counter')).componentInstance;
+  });
+
+  it('should emit valueChange event on counter button click', () => {
+    const valueChangeHandlerSpy = spyOn(hostComponent, 'onValueChange');
+    const counterButton = fixture.debugElement.query(By.css('.counter-button')).nativeElement as HTMLButtonElement;
+
+    expect(valueChangeHandlerSpy).toHaveBeenCalledTimes(0);
+
+    counterButton.click();
+
+    expect(valueChangeHandlerSpy).toHaveBeenCalledTimes(1);
+  });
+});
+```
+
+### Another alternative, ngneat/spectator
+
+If you are willing to add a 3rd party library to the project, [ngneat/spectator](https://github.com/ngneat/spectator) has some useful helpers for setting inputs and triggering CD, even for OnPush components. However, it will change the way you set up tests so it is up to you if you want to go this route or not.
```

Comment:
How do I know am I willing to add it? 🙂  The idea of the Handbook is to be a guide for what to use in projects. I would  either recommend to use this package, or omit it completely. Since I'm hearing about this package for the first time, I would omit it. Otherwise, it should state when should I use this package, and when not.

Reply:
Ok I will remove it.